Index: Default.js
===================================================================
--- Default.js	(revision 15592)
+++ Default.js	(working copy)
@@ -285,7 +285,7 @@
 
 				dx = tickVector.x * taMinorTick.length;
 				dy = tickVector.y * taMinorTick.length;
-				canLabel = c.minMinorStep <= c.minor.tick * c.scale;
+				canLabel = c.minMinorStep <= c.minor.tick * c.bounds.scale;
 				dojo.forEach(t.minor, function(tick){
 					var offset = f(tick.value), elem,
 						x = start.x + axisVector.x * offset,
